Source/BugsnagSession.h

```diff
@@ -9,24 +9,34 @@
 #import <Foundation/Foundation.h>
 
 #import "BugsnagUser.h"
+#import "BugsnagApp.h"
+#import "BugsnagDevice.h"
 
 @interface BugsnagSession : NSObject
 
-- (_Nonnull instancetype)initWithId:(NSString *_Nonnull)sessionId
-                 startDate:(NSDate *_Nonnull)startDate
-                      user:(BugsnagUser *_Nullable)user
-              autoCaptured:(BOOL)autoCaptured;
-
-- (_Nonnull instancetype)initWithDictionary:(NSDictionary *_Nonnull)dict;
-
-- (_Nonnull instancetype)initWithId:(NSString *_Nonnull)sessionId
-                          startDate:(NSDate *_Nonnull)startDate
-                               user:(BugsnagUser *_Nullable)user
-                       handledCount:(NSUInteger)handledCount
-                     unhandledCount:(NSUInteger)unhandledCount;
-
-@property(readonly) NSString *_Nonnull sessionId;
-@property(readonly) NSDate *_Nonnull startedAt;
-@property(readonly) BugsnagUser *_Nullable user;
+@property NSString *_Nonnull id;
+@property NSDate *_Nonnull startedAt;
+@property(readonly) BugsnagApp *_Nonnull app;
+@property(readonly) BugsnagDevice *_Nonnull device;
```

Comment:
I think these two properties can be made private and added to the category on `BugsnagSessionTrackingPayload`

Reply:
These properties are mandated as public in the notifier spec so a user can inspect/mutate them in a callback - otherwise hiding them would be the right approach.